src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -411,7 +412,7 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     completionWork onFailure {
       case e =>
         logger.error(s"Preemption work failed for call ${callKey.tag}! " + e.getMessage, e)
-        scheduleTransition(WorkflowFailed)
+        self ! CheckForWorkflowComplete
```

Comment:
How is that encapsulation different from a method, except that now there's a new message type and handler case?  

The motivation for internal messages here is to guarantee that mutable state is managed only in the actor's single thread.  When a composed future wants to manipulate mutable state, that can't be done on an arbitrary `Future` thread that runs asynchronously to the actor, so the actor messages itself to schedule the manipulation to be done synchronously.